kubernetes-ts-multicloud/aks.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as azure from "@pulumi/azure";
+import * as azuread from "@pulumi/azuread";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import * as random from "@pulumi/random";
+import * as tls from "@pulumi/tls";
+
+// Arguments for an AKS cluster. We use almost all defaults for this example, but the
+// interface could be extended with e.g. agent pool settings.
+export interface AksClusterArgs {
+}
+
+export class AksCluster extends pulumi.ComponentResource {
+    public cluster: azure.containerservice.KubernetesCluster;
+    public provider: k8s.Provider;
+    public staticAppIP: pulumi.Output<string>;
+
+    constructor(name: string,
+                args: AksClusterArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        super("examples:kubernetes-ts-multicloud:AksCluster", name, args, opts);
+
+        const password = new random.RandomString("password", {
+            length: 20,
+            special: true,
+        }, {parent: this}).result;
+        const sshPublicKey = new tls.PrivateKey("sshKey", {
```

Comment:
I would expect ssh pub key to be user-specified, instead of generated. Thoughts?

Reply:
I'll add a note, but did this for easy demo purposes.